pkg/challenges/challenges.go

```diff
@@ -270,14 +275,15 @@ func workflowInfoFromIDToken(token *oidc.IDToken) (map[AdditionalInfo]string, er
 }
 
 func isSpiffeIDAllowed(host, spiffeID string) bool {
-	// Strip spiffe://
-	name := strings.TrimPrefix(spiffeID, "spiffe://")
-
-	// get the host part
-	spiffeDomain := strings.Split(name, "/")[0]
-
-	if spiffeDomain == host {
+	u, err := url.Parse(spiffeID)
+	if err != nil {
+		return false
+	}
+	if u.Scheme != "spiffe" {
+		return false
+	}
+	if u.Hostname() == host {
```

Comment:
I changed this because writing my unit test (`api_test.go`) this doesn't properly handle port numbers at HEAD.